src/components/SubmitLink.tsx

```diff
@@ -3,24 +3,29 @@ import { isLinkDuplicate } from "../functions/isLinkDuplicate";
 import { isLinkValid } from "../functions/isLinkValid";
 import { ListItem } from "../types/ListItem";
 import { UrlType } from "../types/Url";
-
+import { Button } from "@material-ui/core";
 type SubmitProps = UrlType & {
   items: ListItem[];
-  onClick: (event: React.FormEvent<HTMLInputElement>) => void;
+  onClick: (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ) => void | undefined;
 };
 
 export class SubmitLink extends React.Component<SubmitProps> {
   render(): JSX.Element {
     return (
-      <input
-        type="submit"
-        value="Add link"
+      <Button
+        variant="outlined"
+        color="primary"
         disabled={
           !isLinkValid(this.props.url) ||
-          isLinkDuplicate(this.props.url, this.props.items)
+          isLinkDuplicate(this.props.url, this.props.items) ||
+          this.props.url.length >= 2083
```

Comment:
It's good to name numerical constants so that we (or someone else reading the code) don't forget in future what's the purpose of them. Also beware of [this antipattern](https://en.wikipedia.org/wiki/Magic_number_(programming)).